packages/opentelemetry-instrumentation-lancedb/tests/test_query.py

```diff
@@ -0,0 +1,62 @@
+import lancedb
+import pytest
+from opentelemetry.semconv_ai import SpanAttributes
+
+db = lancedb.connect("data/sample-lancedb")
+
+
+@pytest.fixture
+def collection():
+    data = [
+        {"vector": [1.3, 1.4], "item": "fizz", "price": 100.0},
+        {"vector": [9.5, 56.2], "item": "buzz", "price": 200.0},
+    ]
+    yield db.create_table("my_table", data=data)
+    db.drop_table("my_table")
+
+
+def add_data(collection):
+    dataToAdd = [
+        {"vector": [1.3, 1.4], "item": "fizz", "price": 100.0},
+        {"vector": [9.5, 56.2], "item": "buzz", "price": 200.0},
+    ]
+    tbl = db.open_table("my_table")
+    tbl.add(data=dataToAdd)
+
+
+def test_lancedb_add(exporter, collection):
+    exporter.clear()
+    add_data(collection)
+
+    spans = exporter.get_finished_spans()
+    span = next(span for span in spans if span.name == "lancedb.add")
+
+    assert span.attributes.get(SpanAttributes.VECTOR_DB_VENDOR) == "lancedb"
+    assert span.attributes.get(SpanAttributes.VECTOR_DB_OPERATION) == "add"
+    assert span.attributes.get(SpanAttributes.MILVUS_INSERT_DATA_COUNT) == 2
```

Comment:
Hmm... We don't have a generic one for this? :(

Reply:
I don't think so right now, but I'd be happy to discuss how we can genericize VectorDB Operations without losing the small semantic differences between providers.